CRM/Admin/Form/Setting/Smtp.php

```diff
@@ -238,6 +247,8 @@ public function setDefaultValues() {
     if (!$this->_defaults) {
       $this->_defaults = array();
 
+      $this->_defaults['allow_mail_from_logged_in_contact'] = Civi::settings()->get('allow_mail_from_logged_in_contact');
+
```

Comment:
Put this assignment below just before return because at L254 it overrides the default values and the value for this new setting is no longer set as default.